src/Foundatio.Parsers.ElasticQueries/Visitors/ElasticQueryVisitorContext.cs

```diff
@@ -1,18 +1,13 @@
 ﻿using System;
+using System.Collections.Generic;
 using Foundatio.Parsers.LuceneQueries.Visitors;
-using Nest;
-using System.Linq;
-using System.Threading.Tasks;
-using Elasticsearch.Net;
 
 namespace Foundatio.Parsers.ElasticQueries.Visitors {
-    public class ElasticQueryVisitorContext : QueryVisitorContext, IQueryVisitorContextWithIncludeResolver, IQueryVisitorContextWithFieldResolver, IElasticQueryVisitorContext, IQueryVisitorContextWithValidator {
-        public string DefaultTimeZone { get; set; }
+    public class ElasticQueryVisitorContext : QueryVisitorContext, IQueryVisitorContextWithIncludeResolver, IQueryVisitorContextWithFieldResolver, IElasticQueryVisitorContext, IQueryVisitorContextWithValidation {
+        public Lazy<string> DefaultTimeZone { get; set; }
```

Comment:
I'm really not sure what benefit this gives us. Resolving this string property isn't the expensive part it's when we need to do a potential external request to resolve the time zone. That code would then set this property. Which we can't do here as we need to resolve it in an async call.

Reply:
Yeah, this probably should be async, but there is a lot of non-async code all around it and not sure if I can get it to be async.